### PR TITLE
[GNA] Fix accuracy degradation in compact mode

### PR DESCRIPTION
**Pull request by marcinkusm**
### Details:
- set output data lifetime for DelayedCopy layer to maximum possible value
    to avoid reusage of the output buffer by other layers

### Tickets:
- 86342

### CODE REVIEW

**Review by sirzabek — APPROVED** [review]

File: src/plugins/intel_gna/gna_plugin.cpp [hunk 1]
```diff
@@ -892,18 +892,19 @@ void GNAPlugin::LoadNetwork(const CNNNetwork& _network) {
...
     std::unordered_map<std::string, std::vector<InferenceEngine::CNNLayerPtr>> memoryPairs;
     // find all memory layers pairs and mark which one used as outputs
-    uint16_t id = 0;
+    int id = 0;
     for (auto& layer : sortedNet) {
         // set order id for layers to use it in compact mode
+        LayerInfo layerInfo(layer);
         IE_SUPPRESS_DEPRECATED_START
-        layer->userValue.v_int = id++;
+        layer->userValue.v_int = layerInfo.isCopyDelayed() ? std::numeric_limits<int>::max() : id++;
```
[the remark this answers is not in the document]
Comment (marcinkusm) [reply src/plugins/intel_gna/gna_plugin.cpp:902]: comment added.

**Comment by nkogteva** [comment]
Maybe we can use it as workaround. But I have the following questions:

1. Whether problem is still reproducable here https://github.com/openvinotoolkit/openvino/pull/12157 (transformation which inserts copy layers moved to ngraph)
2. Can we have add test whcih reproduces problem?

**Comment by marcinkusm** [comment]
> 
I checked your PR in Release config for linux and issue seems to be not reproducible there. I saw that one pass less is visible.

**Comment by sirzabek** [comment]
> 

1. Even if the issue is not reproducible with the mentioned PR, I think there still might be a problem with other model where the legacy copy inserting pass is used, which is not removed by the PR. Anyway we should develop a generic solution to prevent these type of issues in compact mode.
2. The problem here is that to create a synthetic model which would reproduce the issue we would have to mimic the heuristics used for compact mode. I think the pragmatic solution is to use customer model for testing.

**Review by nkogteva — APPROVED** [review]

**Pull request merged**